cgroup2/manager.go

```diff
@@ -625,56 +627,53 @@ func (c *Manager) Stat() (*stats.Metrics, error) {
 	return &metrics, nil
 }
 
-func getUint64Value(key string, out map[string]interface{}) uint64 {
+func getUint64Value(key string, out map[string]string) uint64 {
 	v, ok := out[key]
 	if !ok {
 		return 0
 	}
-	switch t := v.(type) {
-	case uint64:
-		return t
+	val, err := parseUint(v, 10, 64)
+	if err != nil {
+		return 0
 	}
-	return 0
+	return val
 }
 
-func getPidValue(key string, out map[string]interface{}) uint64 {
+func getPidValue(key string, out map[string]string) uint64 {
 	v, ok := out[key]
 	if !ok {
 		return 0
 	}
-	switch t := v.(type) {
-	case uint64:
-		return t
-	case string:
-		if t == "max" {
-			return math.MaxUint64
-		}
+	if v == "max" {
+		return math.MaxUint64
+	}
+	val, err := parseUint(v, 10, 64)
+	if err != nil {
+		return 0
 	}
-	return 0
+	return val
 }
 
-func readSingleFile(path string, file string, out map[string]interface{}) error {
+func readSingleFile(path string, file string, out map[string]string) error {
 	f, err := os.Open(filepath.Join(path, file))
 	if err != nil {
 		return err
 	}
 	defer f.Close()
-	data, err := io.ReadAll(f)
+
+	// We expect an unsigned 64 bit integer, or a "max" string
+	// in some cases.
+	buf := make([]byte, 32)
```

Comment:
Is this way too low or
https://github.com/containerd/cgroups/pull/278/files#diff-f6fc333a9f621c02b1a9723d8cae6ad3ddd6cdc2501c75cc2a5765bbf01e631cR140 this too high?

Reply:
Yea, I think it's too high. It was originally a brainfart on my part of attributing 64 bit with needing 64 bytes for the character representation in the file, and I found out what the trimming of space was likely for for all of these parsing functions. All of the cgroup files seem to end with a single line feed, and TrimSpace counts this as white space, so it's basically just removing \n for any of the files.  We should only need a buffer of 20 to handle the max uint64, so 32's just for power of 2. I'll open a change to lower the other buffer